sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs

```diff
@@ -920,6 +926,16 @@ public partial class InfluxDbDataFeedSource : Azure.AI.MetricsAdvisor.Models.Dat
         public static bool operator !=(Azure.AI.MetricsAdvisor.Models.IngestionStatusType left, Azure.AI.MetricsAdvisor.Models.IngestionStatusType right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class LogAnalyticsDataFeedSource : Azure.AI.MetricsAdvisor.Models.DataFeedSource
+    {
+        public LogAnalyticsDataFeedSource(string workspaceId, string query) { }
```

Comment:
so a secret is not needed for all cases?

Reply:
Correct. For example, a data source may have public read-only access enabled, making authentication unnecessary. I need to revisit some data sources and check with the service which authentication properties are required and which are not (https://github.com/Azure/azure-sdk-for-net/issues/21616).